### ISSUE TITLE
archetypes.multilingual not working in Plone 4.3.4 using Archetypes

### ISSUE THREAD

**Issue by frisi** [plone/archetypes.multilingual#13]
translating content within a language root folder results in a site error:

```
    Module plone.app.multilingual.browser.utils, line 155, in current_language_name
  AttributeError: 'NoneType' object has no attribute 'get'
  > /home/frisi/instances/aida/src/plone.app.multilingual/src/plone/app/multilingual/browser/utils.py(155)current_language_name()
      154         lang_info = data.get(lang_code)
  --> 155         return lang_info.get('native', None) or lang_info.get('name')
```
### to reproduce:
1. add german and english as site languages
2. create a folder "about us" in Plone/en
3. in the translations dropdown click "create german"

i was using the following versions, but latest 2.x releases will result in the same error

```
plone.app.multilingual               = git https://github.com/plone/plone.app.multilingual.git branch=2.x
archetypes.multilingual              = git https://github.com/plone/archetypes.multilingual.git branch=2.x
```
### analysis

i started looking into this and fixed the way the current language is computed in `archetypes.multilingual`'s implementation of the `ILanguage` adapter
(see the branch https://github.com/plone/archetypes.multilingual/tree/fix-translation)

now invoking "create german" works and i can save a german version of the document.
however i can't navigate between the translations using the portal-languageselector viewlet
(i did not yet check whether the translation relation is not properly set or if it's the @@multilingual-selector view's fault (this one is new to me coming from LinguaPlone)

looking at the tests of both packages (archetypes.multilingual and p.a.multilingual) it looks like p.a.multilingual is not properly tested or even tested at all for archetypes content types.
there should be a test, similar to test_translate_content.robot but using a layer that configures a plonesite using ATContentTypes instead of plone.app.contenttypes.

am i using the wrong versions here, or will i need to put some effort into getting p.a.multilingual to work with archetypes and dexterity content types for my project?

**Comment by saily** [comment]
Hi @frisi, for older Plone sites i use 1.2.1 version. In principal ``plone.app.multilingual`` 2.0 should work for Archetypes, but - to be honest - the last 4 years i did no project with Archetypes.

**Comment by frisi** [comment]
In the meantime i found out, that the guys from IMIO (mainly @bsuttor)  and @mpeeters  seem to have had the same issues and started to work on PRs (here on [archetypes.multlilingual](https://github.com/plone/archetypes.multilingual/pull/10) and on [plone.app.multiligual](https://github.com/plone/plone.app.multilingual/pull/161), too) to fix this.

If I use [their sources.cfg](https://github.com/IMIO/buildout.website/blob/4.3.3/sources.cfg#L42) creating translations for AT objects works fine::

```
archetypes.multilingual               = git https://github.com/IMIO/archetypes.multilingual.git branch=remove-session
plone.app.multilingual                = git https://github.com/IMIO/plone.app.multilingual.git branch=2.x
```

however, the widget in the dialog for linking existing objects as translations (@@add_translations) still does not work (contenttreewidget does not find items of the specified language).
do i need plone.app.widgets to make this work properly?

Using the 1.x branches works fined for archetypes. (uses archetypes.referencebrowserwidget for the add_translations dialog)
```
plone.app.multilingual               = git https://github.com/plone/plone.app.multilingual.git branch=1.x
archetypes.multilingual              = git https://github.com/plone/archetypes.multilingual.git branch=1.x
```

however i'm not sure which (possible important) feature i'll miss. for dexterity, as well as in general (think "real shared content" - what's that exactly and why does one need this? a little explanation in the readme would be helpful. maybe @datakurre can shed some light on this?)
in addition upgrading to the 2.x branches mentioned above results in an error when accessing any content within the plonesite or the ZMI:

```
2015-05-12 23:52:50 WARNING OFS.Uninstalled Could not import class 'createdEvent' from module 'plone.multilingual.subscriber'
2015-05-12 23:52:50 WARNING OFS.Uninstalled Could not import class 'update_on_modify' from module 'plone.multilingual.subscriber'
2015-05-12 23:52:50 ERROR ZODB.Connection Couldn't load state for 0x8e7e
Traceback (most recent call last):
  File "eggs/ZODB3-3.10.5-py2.7-linux-x86_64.egg/ZODB/Connection.py", line 860, in setstate
    self._setstate(obj)
  File "eggs/ZODB3-3.10.5-py2.7-linux-x86_64.egg/ZODB/Connection.py", line 914, in _setstate
    self._reader.setGhostState(obj, p)
  File "eggs/ZODB3-3.10.5-py2.7-linux-x86_64.egg/ZODB/serialize.py", line 613, in setGhostState
    obj.__setstate__(state)
  File "eggs/zope.component-3.9.5-py2.7.egg/zope/component/persistentregistry.py", line 40, in __setstate__
    self._createLookup()
  File "eggs/zope.interface-3.6.7-py2.7-linux-x86_64.egg/zope/interface/adapter.py", line 91, in _createLookup
    self._v_lookup = self.LookupClass(self)
  File "eggs/zope.interface-3.6.7-py2.7-linux-x86_64.egg/zope/interface/adapter.py", line 439, in __init__
    self.init_extendors()
  File "eggs/zope.interface-3.6.7-py2.7-linux-x86_64.egg/zope/interface/adapter.py", line 476, in init_extendors
    self.add_extendor(p)
  File "eggs/zope.interface-3.6.7-py2.7-linux-x86_64.egg/zope/interface/adapter.py", line 480, in add_extendor
    for i in provided.__iro__:
AttributeError: type object 'ITranslationFactory' has no attribute '__iro__'
```



to sum up:

it looks like using 1.x for a start and upgrade to 2.x later (if i need better support for dexterity or "real shared content") is not working out of the box because:

A) it will not work with archetypes.
is there a specific reason why the IMIO pull request have not been merged or at least commented on?

B) the ITranslationFactory error mentioned above.
this might be something @mpeeters  wanted to fix with https://github.com/IMIO/plone.app.multilingual/commit/412cd887d80d05cc121f5060d27313f25cee4b85 or am I completely wrong?

**Comment by mck9** [comment]
I don't know if it is related, but I can't even set the language. The update_language languages dropdown is empty. 

Plone 4.3.4
plone.app.multilingual[archetypes]==2.0.0

**Comment by keul** [comment]
I found  the same issue on Plone 4.3.6. Pull request will follow.
